File: Problem of the day/Kadane's Algorithm.java
//{ Driver Code Starts
// Initial Template for Java
import java.io.*;
import java.util.*;

class Main {

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(br.readLine().trim()); // Inputting the testcases
        while (t-- > 0) {

            String line = br.readLine();
            String[] tokens = line.split(" ");

            // Create an ArrayList to store the integers
            ArrayList<Integer> array = new ArrayList<>();

            // Parse the tokens into integers and add to the array
            for (String token : tokens) {
                array.add(Integer.parseInt(token));
            }

            int[] arr = new int[array.size()];
            int idx = 0;
            for (int i : array) arr[idx++] = i;

            Solution obj = new Solution();

            // calling maxSubarraySum() function
            System.out.println(obj.maxSubarraySum(arr));
        }
    }
}

// } Driver Code Ends

class Solution {
    // Function to find the sum of contiguous subarray with the maximum sum.
    int maxSubarraySum(int[] arr) {
        // Initialize maxSum with the smallest integer value and currentSum as 0.
        int maxSum = Integer.MIN_VALUE;
        int currentSum = 0;

        for (int i = 0; i < arr.length; i++) {
            // Add current element to currentSum.
            currentSum += arr[i];
            
            // Update maxSum if the currentSum is greater.
            if (currentSum > maxSum) {
                maxSum = currentSum;
            }

            // If currentSum drops below 0, reset it to 0 (start a new subarray).
            if (currentSum < 0) {
                currentSum = 0;
            }
        }

        return maxSum;
    }
}

File: Problem of the day/Kadane's Algorithm.py
class Solution:
    ## Complete this function
    # Function to find the sum of contiguous subarray with maximum sum.
    def maxSubArraySum(self, arr):
        # Initialize variables
        max_sum = float('-inf')  # To handle negative arrays
        current_sum = 0
        
        for num in arr:
            # Add current element to current_sum or start a new subarray
            current_sum = max(num, current_sum + num)
            # Update max_sum if current_sum is larger
            max_sum = max(max_sum, current_sum)
        
        return max_sum


#{ 
 # Driver Code Starts
#Initial Template for Python 3

import math


def main():
    T = int(input())
    while (T > 0):

        arr = [int(x) for x in input().strip().split()]

        ob = Solution()

        print(ob.maxSubArraySum(arr))

        T -= 1


if __name__ == "__main__":
    main()

# } Driver Code Ends